src/test/java/org/broadinstitute/hellbender/utils/config/ConfigIntegrationTest.java

```diff
@@ -67,54 +109,94 @@ public String getTestedClassName() {
+
+            // Get the config properties:
+            final LinkedHashMap<String, Object> configMap = ConfigFactory.getConfigMap(config, false);
+
+            // Compare the configuration properties and assert their equality:
+            for ( final Map.Entry<String, Object> entry : configMap.entrySet() ) {
+                Assert.assertEquals(configProperties.getProperty(entry.getKey()), String.valueOf(entry.getValue()));
+            }
+
+            // Compare the system properties and assert their equality:
+            for ( final String sysPropKey : systemPropertiesToQuery.keySet() ) {
+                Assert.assertEquals(systemPropertiesPostToolRun.getProperty(sysPropKey), systemPropertiesToQuery.get(sysPropKey));
+            }
+
+            // Make sure they have the same size:
+            Assert.assertEquals(configProperties.size(), configMap.size());
 
-        // Run our command:
-        runCommandLine(args.getArgsArray());
-
-        // Now we get to read in the file's contents and compare them to our config settings:
-        final Properties properties = new Properties();
-        try ( final FileInputStream inputStream = new FileInputStream(tmpFile) ) {
-            properties.load(inputStream);
-        }
-        catch (final Exception ex) {
-            throw new GATKException("Test error!", ex);
         }
-
-        // Create a config object to compare with the properties:
-        final GATKConfig config = ConfigFactory.getInstance().createConfigFromFile(configFilePath.toString(), GATKConfig.class);
-
-        // Get the config properties:
-        final LinkedHashMap<String, Object> configMap = ConfigFactory.getConfigMap(config, false);
-
-        // Compare the properties and assert their equality:
-        for ( final Map.Entry<String, Object> entry : configMap.entrySet() ) {
-            Assert.assertEquals( properties.getProperty(entry.getKey()), String.valueOf(entry.getValue()) );
-        }
-
-        // Make sure they have the same size:
-        Assert.assertEquals( properties.size(), configMap.size() );
-
         // =========================================================================================================
         // Now we have to reset our system options:
-
-        // Run with normal options to reset any transient testing changes:
-        final GATKConfig defaultConfig = ConfigFactory.getInstance().createConfigFromFile(null, GATKConfig.class);
-
-        // Get our system properties and what they should be:
-        final Map<String, Object> systemConfigOptions = ConfigFactory.getConfigMap(defaultConfig, true);
-
-        // Reset our system properties:
-        for ( final Map.Entry<String, Object> entry : systemConfigOptions.entrySet() ) {
-            System.setProperty(entry.getKey(), entry.getValue().toString());
+        finally {
+            // Clear our system properties:
+            final List<Object> keyList = new ArrayList<>(System.getProperties().keySet());
```

Comment:
Does System.getProperties().clear() not work to clear them all?

Reply:
I'm not sure.  I'm using `System.clearProperty` because that's the interface that specifically says it will clear a specific key.

Reply:
reasonable